PayloadPkg/FirmwareUpdate/FirmwareUpdate.c

```diff
@@ -1220,8 +1221,13 @@ ApplyFwImage (
     Status = UpdateSystemFirmware(ImageHdr);
     *ResetRequired = TRUE;
   } else if (CompareGuid(&ImageHdr->UpdateImageTypeId, &gCsmeFWUpdateImageFileGuid) == TRUE) {
-    Status = UpdateCsme(CapImage, CapImageSize, ImageHdr);
-    *ResetRequired = TRUE;
+    if (FeaturePcdGet(PcdCsmeUpdateEnabled) == 1) {
+      CsmeUpdateInData = InitCsmeUpdInputData();
+      if (CsmeUpdateInData != NULL) {
+        Status = UpdateCsme(CapImage, CapImageSize, CsmeUpdateInData, ImageHdr);
+      }
+    }
+    *ResetRequired = FALSE;
```

Comment:
Is 'else {}' missing? or No need to reset all the time?

Reply:
Right, corrected code